### PR TITLE
Hang around lured pokestops

### PR DESCRIPTION
**Pull request by jabbink**
**Fixed issue:** #54

**Changes made:**

* Camp near `x` lured Pokestops in range

### CODE REVIEW

File: src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt [hunk 1]
```diff
@@ -42,13 +44,22 @@ class ProcessPokestops(var pokestops: MutableCollection<Pokestop>) : Task {
...
         if (settings.lootPokestop) {
             val loot = LootOneNearbyPokestop(sortedPokestops, lootTimeouts)
             bot.task(loot)
         }
+        if (settings.campLurePokestop > 0) {
+            // TODO: grab that 70 from DOWNLOAD_SETTINGS
+            val luresInRange = sortedPokestops.filter { it.inRange(ctx.api, 70.0) && it.hasLure() }.size
+            if (luresInRange >= settings.campLurePokestop) {
+                Log.green("$luresInRange lures in range, pausing")
+                ctx.api.map.getMapObjects(3)
```
Comment (jabbink) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt]: @grover-c13 am I correct to assume that the API won't return new `CatchablePokemon` without this call (aka the cache is never invalidated by a timer)?

File: src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt [hunk 2]
```diff
@@ -42,13 +44,22 @@ class ProcessPokestops(var pokestops: MutableCollection<Pokestop>) : Task {
...
 
         if (settings.lootPokestop) {
             val loot = LootOneNearbyPokestop(sortedPokestops, lootTimeouts)
             bot.task(loot)
         }
+        if (settings.campLurePokestop > 0) {
+            // TODO: grab that 70 from DOWNLOAD_SETTINGS
+            val luresInRange = sortedPokestops.filter { it.inRange(ctx.api, 70.0) && it.hasLure() }.size
+            if (luresInRange >= settings.campLurePokestop) {
+                Log.green("$luresInRange lures in range, pausing")
```
Comment (jabbink) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt]: Spams the console...

File: src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt [hunk 3]
```diff
@@ -42,13 +43,22 @@ class ProcessPokestops(var pokestops: MutableCollection<Pokestop>) : Task {
...
 
         if (settings.lootPokestop) {
             val loot = LootOneNearbyPokestop(sortedPokestops, lootTimeouts)
             bot.task(loot)
         }
+        if (settings.campLurePokestop > 0) {
+            val luresInRange = sortedPokestops.filter {
+                it.inRangeForLuredPokemon() && it.fortData.hasLureInfo()
+            }.size
+            if (luresInRange >= settings.campLurePokestop) {
```
Comment (CosmoRing) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt]: @jabbink shouldn't you check if catchPokemon in settings is true? If it's not there isn't any way to catch the lured pokemons.

Comment (jabbink) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt]: Good one, but luckily won't be a problem for now as the default setting of this camp thing is OFF and people that set catchPokemon to false can just disable this option (let it disabled)

Comment (CosmoRing) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/ProcessPokestops.kt]: I know that, but i'm already imagining people complaining not getting any pokemon while on lured pokestops or people saying they get "stuck" on pokestops. 

Anyway it was just a warning :)

**Pull request merged**